robusta_krr/core/integrations/prometheus/metrics/cpu_metric.py

```diff
@@ -2,9 +2,17 @@
 
 from .base_metric import bind_metric
 from .base_filtered_metric import BaseFilteredMetricLoader
+from robusta_krr.core.models.objects import K8sObjectData
 
 
 @bind_metric(ResourceType.CPU)
 class CPUMetricLoader(BaseFilteredMetricLoader):
-    def get_query(self, namespace: str, pod: str, container: str) -> str:
-        return f'sum(irate(container_cpu_usage_seconds_total{{namespace="{namespace}", pod="{pod}", container="{container}"}}[5m])) by (container, pod, job)'
+    def get_query(self, object: K8sObjectData) -> str:
+        pods_selector = "|".join(pod.name for pod in object.pods)
```

Comment:
much better

Reply:
if we do this
 (
            'sum(irate(container_cpu_usage_seconds_total{}[5m])) by (container, pod, job)'
        )
     
     we will get all the metrics for every container and pod and can filter it on our end, it might be more efficient /  faster than calling query_prometheus for every deployment,DS,ss, job ..., WDYT?
     (same with memory)

Reply:
then again it might be allot of data that we get in the return result, it might be worth checking if the tradeoff is worth it

Reply:
Even tho it might be more efficient, in fact it currently is fast (mostly because we are doing that in parallel)
Also, we need the query to represent the resulted data for making graphs in Robusta UI.